vm-ospatching/azuredeploy.json

```diff
@@ -0,0 +1,236 @@
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "West US",
+                "metadata": {
+                    "description": "The location where the Virtual Machine will be deployed."
+                }
+        },
+        "adminUsername": {
+            "type": "string",
+                "metadata": {
+                   "description": "Username for the Virtual Machine."
+                }
+        },
+        "adminPassword": {
+            "type": "securestring",
+                "metadata": {
+                    "description": "Password for the Virtual Machine."
+                }
+        },
+        "dnsNameForPublicIP": {
+            "type": "string",
+                "metadata": {
+                      "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+                }
+        },
+        "ubuntuOSVersion": {
+            "type": "string",
+            "defaultValue": "14.04.2-LTS",
+                "metadata": {
+                      "description": "The Ubuntu version for deploying the OS patching extension. This will pick a fully patched image of this given Ubuntu version. Allowed values: 14.04.2-LTS, 14.04-DAILY, 15.04, 14.10."
+                 },
+                "allowedValues": [
+                    "14.04.2-LTS",
+					          "14.04-DAILY",
+                    "15.04",
+                    "14.10"
+                ]
+        },
+        "rebootAfterPatch":{
+            "type" : "string",
+            "defaultValue" : "Auto",
+            "metadata": {
+              "description": "The reboot behavior after patching."
+            }
+        },
+        "dayOfWeek":{
+            "type" : "string",
+            "defaultValue" : "Sunday|Wednesday",
+            "metadata": {
+              "description": "The patching date (of the week)You can specify multiple days in a week."
+            }
+        },
+        "startTime":{
+            "type" : "string",
+            "defaultValue" : "03:00",
+            "metadata": {
+              "description": "Start time of patching."
+            }
+        },
+        "category":{
+            "type" : "string",
+            "defaultValue" : "ImportantAndRecommended",
+            "metadata": {
+              "description": "Type of patches to install."
+            }
+        }
+    },
+    "variables": {
+        "imagePublisher": "Canonical",
+        "imageOffer": "UbuntuServer",
+        "OSDiskName": "osdiskforospatching",
+        "nicName": "myVMNic",
+        "extensionName": "OSPatching",
+        "addressPrefix": "10.0.0.0/16",
+        "subnetName": "Subnet",
+        "subnetPrefix": "10.0.0.0/24",
+        "storageAccountType": "Standard_LRS",
+        "publicIPAddressName": "myPublicIP",
+        "publicIPAddressType": "Dynamic",
+        "vmStorageAccountContainerName": "vhds",
+        "vmName": "MyOSPatchingVM",
+        "vmSize": "Standard_D1",
```

Comment:
Make this a parameter with a default. Then the user has a chance to change the size before deployment